### PR TITLE
Ac 82 create order/order items models/schemas

### LINKED ISSUE #82: feat(f6): Create Order and Order-Items models and schemas
- Creating an order model and schema so the user can make a merch/game order
- Creating an order items-model and schema to be able to hold an array of products that will be shown in the order

### PR DESCRIPTION
**Pull request by sirpuffin1**
1. Create order/order-item model
2. Create order/order-item schema

Closes #82

### CODE REVIEW

**Comment by sirpuffin1** [comment]
Really hoping I did this right

**Review by ChelleAmon — COMMENTED** [review]
The order schema is referenced only to merch. Are you going to create a separate Order collection for games?

**Comment by Aman-codde** [comment]
> The order schema is referenced only to merch. Are you going to create a separate Order collection for games?

that's a good point that we can ask Vinson maybe cuz if a customer is ordering game and a merch, he/she should be able to purchase both and same question is for cart also.

File: src/shared/models/order-items.model.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { Merch } from "./merch.model";
+export interface orderItems {
+    quantity: number,
+    product: string | Merch
```
Comment (Aman-codde) [inline src/shared/models/order-items.model.ts:4]: I think product shape could be **product: Merch | Game** to order both merch and games but I am not sure if that's possible, something we can discuss maybe

File: src/shared/models/order.model.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { orderItems } from "./order-items.model";
+import { User } from "./user.model";
+
+export interface Order {
+    _id?: string,
+    orderItems: orderItems[],
+    shippingAddress1: string,
```
Comment (Aman-codde) [inline src/shared/models/order.model.ts]: the shape of shippingAddress property can also be of **shippingAddress1: ShippingAddress** where 'ShippingAddress' can be an interface in the same model which holds properties like city, zip, country, phone inside it

File: src/shared/models/order-items.model.ts (hunk 1 above)
Comment (brycejjacks) [reply src/shared/models/order-items.model.ts:4]: I think a little differently than Aman on this one, I say we do not have an order for games and only for merch. So this is fine(string if we are passing the id | merch if populated ), and we purchase each game individually then add it straight onto the user or in a separate collections similar to how we are doing friends

File: src/shared/models/order.model.ts (hunk 2 above)
Comment (brycejjacks) [reply src/shared/models/order.model.ts]: I 100% agree with Aman on this one tho lol.. stop being lazy

File: src/server/schemas/order-items.schema.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import  mongoose from 'mongoose';
+import { orderItems } from '../../shared/models/order-items.model';
+const {Schema, model} = mongoose
+
+const orderItemSchema = new Schema<orderItems>({
```
Comment (brycejjacks) [inline src/server/schemas/order-items.schema.ts:5]: i really dont like the name of this schema lol ... inventory maybe .. but that is me being picky

File: src/shared/models/order.model.ts (hunk 2 above)
Comment (sirpuffin1) [reply src/shared/models/order.model.ts]: I honestly didn't even think to do that good idea lol

File: src/shared/models/order-items.model.ts (hunk 1 above)
Comment (sirpuffin1) [reply src/shared/models/order-items.model.ts:4]: Yes exactly but we can talk about it more tomorrow morning

File: src/server/schemas/order.schema.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+const {Schema, model} = mongoose
+
+
+const orderSchema = new Schema<Order>({
+    orderItems: [{
+        type: mongoose.Schema.Types.ObjectId,
+        ref: 'OrderItem',
+        required: true
+      }],
+      shippingAddress1: {
```
Comment (brycejjacks) [inline src/server/schemas/order.schema.ts]: if you decide to move everything into a ShippingAddress interface, this page might get a bit tricky i would try `ShippingAddress: Schema.Types.Mixed `

**Review by brycejjacks — APPROVED** [review]
im looking for more blocking conversations to start as we speak.

File: src/server/schemas/order.schema.ts (hunk 4 above)
Comment (sirpuffin1) [reply src/server/schemas/order.schema.ts]: I'll take that into consideration thanks bryce

**Pull request merged**